### PR TITLE
feat:[PLG-360] AG "all-resources" should be created by the installer …

### PR DESCRIPTION
**Pull request by akashpaladin**
…

# Description

AG "all-resources" should be created by the installer automatically

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Made some Major improvements
We will now only hit 1 time to ES for updating asset groups
For any new targetTypes, we will only need to update asset group once

Testing code for 3 new target type

Using old code it took 7 mins to update

using latest change it took approximately 50 seconds.

Old system duration:
Start time: 21:01:30
End time: 21:08:30

To find the duration, subtract the start time from the end time:

21:08:30 - 21:01:30 = 7 minutes

So, the old system ran for 7 minutes.

New system duration:
Start time: 23:22:39
End time: 23:23:23

Again, to find the duration, subtract the start time from the end time:

23:23:23 - 23:22:39 = 44 seconds

So, the new system ran for 44 seconds.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My commit message/PR follows the contribution guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class AssetGroupManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AssetGroupManager.class);
+    private static final String DATE_FORMAT = "MM/dd/yyyy HH:mm";
+    private static final String EMPTY_STRING = "";
```
Comment (mkinchin-paladin) [inline jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java:42]: Please don't do this... Just use "".

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Minor comment. Requesting changes just to see test results

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+            "\"alias\": \"%s\"}}]}";
+    private static final String FILTER_TEMPLATE = "\"filter\": %s,";
+    private static final String ES_ATTRIBUTE_TAG = "tags.";
+    private static final String ES_ATTRIBUTE_KEYWORD = ".keyword";
+    private static final String DISTINCT_DATA_SOURCES_QUERY_TEMPLATE = "SELECT DISTINCT dataSourceName FROM cf_Target";
+    private static final String CHECK_PLUGIN_ENABLED_QUERY_TEMPLATE =
+            "SELECT `value` FROM pac_config_properties WHERE cfkey = '%s.enabled'";
+    private static final String CRITERIA_DETAILS_QUERY_TEMPLATE =
+            "SELECT attributeName, attributeValue FROM cf_AssetGroupCriteriaDetails WHERE groupId = '%s'";
+    private static final Map<String, List<String>> dbCache = new HashMap<>();
```
Comment (mkinchin-paladin) [inline jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java:70]: Please add test results how faster this Manager class became after adding caching.

**Review by plyubich — DISMISSED** [review]

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java (hunk 1 above)
Comment (akashpaladin) [reply jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java]: ok

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java (hunk 2 above)
Comment (akashpaladin) [reply jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java:68]: added

**Review by mkinchin-paladin — APPROVED** [review]
LGTM

**Pull request merged**